components/MapComponent.tsx

```diff
@@ -23,83 +32,142 @@ interface Props<T> {
+  const circleLatitude1 = lat + degreesToRadians(radius / earthRadius)
+  const circleLongitude0 =
+    long + degreesToRadians(-radius / earthRadius) / degreesToRadians(lat)
+  const circleLongitude1 =
+    long + degreesToRadians(radius / earthRadius) / degreesToRadians(lat)
+  const bottomCoord = {
+    latitude: circleLatitude0,
+    longitude: long
+  }
+  const leftCoord = {
+    latitude: lat,
+    longitude: circleLongitude0
+  }
+  const rightCoord = {
+    latitude: circleLatitude1,
+    longitude: long
+  }
+  const topCoord = {
+    latitude: lat,
+    longitude: circleLongitude1
+  }
+  return [bottomCoord, leftCoord, rightCoord, topCoord]
 }
 
 // Map view component itself
-export function MapComponent<T extends mapMarker> ({
+export function MapComponent<T extends MapMarker> ({
   initialRegion,
   markers,
-  extractKey,
-  containStyle,
+  customizers,
+  currentSelectedMarker,
+  initialRadius,
   mapStyle,
   movementSettings
 }: Props<T>) {
   // Map references
   const mapRef = React.useRef<MapView | null>(null)
-  const [selectedMarker, setSelectedMarker] = useState<String | undefined>()
-
   const markerData = markers.map((markedMarker) => {
-    const key = extractKey(markedMarker)
     return {
       markedMarker,
-      isSelected: key === selectedMarker
+      isSelected: markedMarker.key === currentSelectedMarker
     }
   })
 
-  // Return the markers so that they appear on the map
+  // Return the markers/circles so that they appear on the map
   const mapMarkerCreations = () => {
-    return markerData.map((marker) => (
-      <Marker
-        key={marker.markedMarker.key}
-        title={marker.markedMarker.title}
-        coordinate={{
-          latitude: marker.markedMarker.location.latitude,
-          longitude: marker.markedMarker.location.longitude
-        }}
-        pinColor={"blue"}
-        onPress={() => {
-          onMarkerClick(
-            marker.markedMarker.location.latitude,
-            marker.markedMarker.location.longitude
-          )
-          onSelected(marker.markedMarker)
-        }}
-      />
-    ))
+    return markerData.map((marker) => {
+      const createdCircle = checkCircleKey(marker)
+      return (
+        <>
+          <Marker
+            key={marker.markedMarker.key}
+            title={marker.markedMarker.title}
+            coordinate={{
+              latitude: marker.markedMarker.location.latitude,
+              longitude: marker.markedMarker.location.longitude
+            }}
+            pinColor={customizationCreation(marker.markedMarker).color}
+            onPress={() => {
+              if (marker.isSelected) {
+                centerMapOnMarker(
+                  marker.markedMarker.location.latitude,
+                  marker.markedMarker.location.longitude,
+                  initialRadius.radius
+                )
+              }
+            }}
+          />
+          {createdCircle}
+        </>
+      )
```

Comment:
We should probably make this more like `FlatList` and call a `renderMarker` prop so that we can use any JSX here which you would call inside the Marker component (and have that `renderMarker` function deal with the circle internally), but we'll worry about it later.

See: https://github.com/react-native-maps/react-native-maps/blob/master/docs/marker.md#children-components

Reply:
We can consider that in the future if we need to render different kinds of markers besides events, but as of right now I think it's safe to assume all events will have the same style of marker.